org.eclipse.xtext/src/org/eclipse/xtext/build/Indexer.java

```diff
@@ -213,21 +221,32 @@ public URI getURI() {
 	protected List<IResourceDescription.Delta> getDeltasForDeletedResources(BuildRequest request,
 			ResourceDescriptionsData oldIndex, BuildContext context) {
 		List<IResourceDescription.Delta> deltas = new ArrayList<>();
-		for (URI deleted : request.getDeletedFiles()) {
-			IResourceServiceProvider resourceServiceProvider = context.getResourceServiceProvider(deleted);
-			if (resourceServiceProvider != null) {
-				operationCanceledManager.checkCanceled(context.getCancelIndicator());
-				IResourceDescription oldDescription = oldIndex != null ? oldIndex.getResourceDescription(deleted)
-						: null;
-				if (oldDescription != null) {
-					DefaultResourceDescriptionDelta delta = new DefaultResourceDescriptionDelta(oldDescription, null);
-					deltas.add(delta);
+		if (oldIndex != null) {
+			for (URI deleted : request.getDeletedFiles()) {
+				IResourceServiceProvider resourceServiceProvider = context.getResourceServiceProvider(deleted);
+				if (resourceServiceProvider != null) {
```

Comment:
I have kept the check, even though I does not know what it is for. The resourceServiceProvider is not used, so from my naive point of view, it could be removed, the get and the null check could be removed, right?

Reply:
the request might be filled by outside tool